### PR TITLE
Sponsorship review ux

### PR DESCRIPTION
**Pull request by berinhard**
This PR enables admin users to Reject/Approve sponsorship applications;

**Reject/Approve buttons**
Screenshot from 2020-11-20 13-13-53

**Confrim approve form**
Screenshot from 2020-11-20 13-13-02

**Confirm reject form**
Screenshot from 2020-11-20 13-12-56

### CODE REVIEW

File: templates/sponsors/partials/full_sponsorship.txt
Comment (berinhard) [inline templates/sponsors/partials/full_sponsorship.txt:16]: I realized that the new address fields are missing herein the `full_sponsorship.txt` =S

**Comment by berinhard** [comment]
Print screens of detailed mail/billing address information:

**sponsorship admin detail**
Screenshot from 2020-11-24 09-49-25

**full sponsorship partial**
Screenshot from 2020-11-24 09-49-59

File: templates/sponsors/partials/full_sponsorship.txt
Comment (berinhard) [reply templates/sponsors/partials/full_sponsorship.txt:16]: Fixed in 
be6273c

File: sponsors/models.py
Comment (ewdurbin) [inline sponsors/models.py]: may want to include start/end date in the string representation, it won't be an issue for first term (initial launch) but will be important for differentiating down the line.

File: sponsors/models.py
Comment (ewdurbin) [inline sponsors/models.py:332]: we probably want to ensure that a valid start/end date was provided as part of the approval process. this should be enforced, as we'll need it for generating Statement of Work.

**Review by ewdurbin — APPROVED** [review]
All looks great. I think there is some business logic I need to verify with sponsorship team, but that shouldn't block us from merging this to move things along.

**Comment by ewdurbin** [comment]
For the purposes of shipping this, I'd like to see how we can display values that need to be provided before approval via admin on the confirmation screen... is that possible? Can we embed a (potentially dynamic) form?

File: sponsors/models.py
Comment (berinhard) [reply sponsors/models.py]: Fixed in d134d3c

File: sponsors/models.py
Comment (berinhard) [reply sponsors/models.py:332]: Totally! I'll update this method to do so, but will do in the statement of work's branch, is it ok? I've changed how the accept view works over there, so it'll be better to change this from a more up to date code.

Comment (ewdurbin) [reply sponsors/models.py:332]: Makes sense! Thank you.

**Review by ewdurbin — APPROVED** [review]

**Comment by berinhard** [comment]
About the dynamic form in the admin, sure we can do that =)

Currently, the approve view is a regular form view which expects a valid post to proceed. Currently there's no form because there's no data. I'll add a simple form for now with with fields for the `start_date` and `end_date`. But, as said [here](https://github.com/python/pythondotorg/pull/1684#discussion_r539513282), I'll base myself in the most recent code base.

**Pull request merged**
